### PR TITLE
nc-news-6

### PR DESCRIPTION
**Pull request by SergMo**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -122,4 +130,58 @@ describe('GET /api/articles', () => {
...
+				)
+			})
+	});
+	test('comments should be served with the most recent comments first', () => {
+		return request(app)
+			.get('/api/articles/1/comments')
+			.expect(200)
+			.then((response) => {
+				const comments = response.body.comments;
+				console.log(comments);
```
Comment (2202Hannah) [inline __tests__/app.test.js]: Please remove this console.log

File: __tests__/app.test.js [hunk 2]
```diff
@@ -122,4 +130,58 @@ describe('GET /api/articles', () => {
...
+				expect(comments).toBeSortedBy('created_at', { descending: true });
+			});
```
Comment (2202Hannah) [inline __tests__/app.test.js:178]: Nice use of the toBeSortedBy matcher here

File: __tests__/app.test.js [hunk 3]
```diff
@@ -122,4 +130,58 @@ describe('GET /api/articles', () => {
...
+	});
+
+	test('GET:404 should handle article not found', () => {
+		return request(app)
+			.get('/api/articles/999/comments')
+			.expect(404)
+			.then((response) => {
+				expect(response.body.message).toBe('Article not found');
+			});
+	});
```
Comment (2202Hannah) [inline __tests__/app.test.js:197]: Please add a test for when the article ID is valid but it has no comments and so responds with an empty array

File: __tests__/app.test.js [hunk 4]
```diff
@@ -122,4 +130,58 @@ describe('GET /api/articles', () => {
...
+describe('GET /api/articles/:article_id/comments', () => {
+
+	test('GET:200 should return an array of comments for the given article_id', () => {
+		return request(app)
+			.get('/api/articles/1/comments')
+			.expect(200)
+			.then((response) => {
+				const comments = response.body.comments;
+				expect(comments).toEqual(expect.any(Array));
+
```
Comment (2202Hannah) [inline __tests__/app.test.js:145]: Please make sure that you also check that the length of the array is as expected

File: controllers/comment.controller.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+	Promise.all([
+		selectArticleById(article_id),
+		selectCommentsByArticleId(article_id)
+	])
+
+		.then(([article, comments]) => {
+			res.status(200).send({ comments });
+		})
+		.catch((err) => {
```
Comment (2202Hannah) [inline controllers/comment.controller.js]: This catch block isn't necessary here as the `article not found` custom error comes from the Promise.reject in the selectArticleById() controller will already be handled by your error handlers.

**Review by 2202Hannah — APPROVED** [review]
Hi Serg, really nice work on this ticket. Please see my comments and make the changes to your code. You can then merge this branch to main when you are ready.

File: __tests__/app.test.js (hunk 1 above)
Comment (SergMo) [reply __tests__/app.test.js]: done

File: __tests__/app.test.js (hunk 3 above)
Comment (SergMo) [reply __tests__/app.test.js:197]: done

File: __tests__/app.test.js (hunk 4 above)
Comment (SergMo) [reply __tests__/app.test.js:145]: done

File: controllers/comment.controller.js (hunk 5 above)
Comment (SergMo) [reply controllers/comment.controller.js]: done
